chain/network/src/peer.rs

```diff
@@ -273,16 +273,32 @@ impl Peer {
                     genesis_id,
                     height,
                     tracked_shards,
+                    archival,
                 }) => {
-                    let handshake = HandshakeV2::new(
-                        act.protocol_version,
-                        act.node_id(),
-                        act.peer_id().unwrap(),
-                        act.node_info.addr_port(),
-                        PeerChainInfo { genesis_id, height, tracked_shards },
-                        act.edge_info.as_ref().unwrap().clone(),
-                    );
-                    act.send_message(PeerMessage::HandshakeV2(handshake));
+                    let handshake = match act.protocol_version {
+                        36..=37 => PeerMessage::Handshake(Handshake::new(
```

Comment:
It is better to use `36..=PROTOCOL_VERSION` in case `PROTOCOL_VERSION` needs to be increased.